modules: lvgl: Create input pointer automatically from chosen touch

faxe1008 opened the pull request:
Having the need to add a pseudo input device for LVGL in every board that has a touch controller onboard is unfortunate.
Instead create a pointer input device for boards where the `zephyr,touch` node is present. Move any modifications to the pointer with `invert-x,y` etc. into the touch controller itsself, so other non LVGL samples also benefit from the corrected coordinate reporting behaviour.

faxe1008 committed:
modules: lvgl: Add LV_Z_POINTER_FROM_CHOSEN_TOUCH option

Adds a Kconfig option LV_Z_POINTER_FROM_CHOSEN_TOUCH which automatically
creates a lvgl input pointer device from the chosen zephyr,touch node.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 committed:
boards: Remove lvgl_input_pointer where chosen zephyr,touch is present

With the new LV_Z_POINTER_FROM_CHOSEN_TOUCH most boards do not need to
specify a dedicated pseudo device anymore.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

faxe1008 committed:
tests: misc: check_init_priorities: Update reference init level list

After removing the `zephyr,lvgl-pointer-input` device the init level
entries have shifted by one entry. Update them accordingly.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>

JarmouniA commented on modules/lvgl/Kconfig.input:
```diff
@@ -4,11 +4,33 @@
+DT_CHOSEN_ZEPHYR_DISPLAY := zephyr,display
+
+config LV_Z_POINTER_FROM_CHOSEN_TOUCH
+	bool "Create lvgl pointer from chosen,zephyr-touch"
```
```suggestion
	bool "Create LVGL pointer from DT 'zephyr,touch' chosen device"
```

JarmouniA commented on modules/lvgl/Kconfig.input:
```diff
@@ -4,11 +4,33 @@
 
 menu "Input device settings"
 
+DT_CHOSEN_ZEPHYR_TOUCH := zephyr,touch
```
```suggestion
DT_CHOSEN_Z_TOUCH := zephyr,touch
```
Seems to be the predominant naming throughout the tree, non-blocking.
https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr+DT_CHOSEN_Z_&type=code

JarmouniA commented on modules/lvgl/Kconfig.input:
```diff
@@ -4,11 +4,33 @@
+	default y if !DT_HAS_ZEPHYR_LVGL_POINTER_INPUT_ENABLED
+	help
+	  Enable creation of LVGL pointer input device
+	  based on the zephyr-touch chosen node.
```
```suggestion
	  based on 'zephyr,touch' chosen property in devicetree.
```

JarmouniA commented on modules/lvgl/Kconfig.input:
```diff
@@ -4,11 +4,33 @@
+	help
+	  Process input events of the pointer device created
+	  from the zephyr,touch chosen node and link it to the
+	  zephyr,display node.
```
```suggestion
	  'zephyr,display' chosen device.
```

faxe1008 committed:
doc: dts: Expand explanation of `zephyr,touch` chosen node

Add explanation of the `CONFIG_LV_Z_POINTER_FROM_CHOSEN_TOUCH` interaction
with the `zephyr,touch` chosen node.

Signed-off-by: Fabian Blatz <fabianblatz@gmail.com>
